### PR TITLE
Updated dict type

### LINKED ISSUE #41: remove redundancy from dict type and support new features
The config.yaml will store the following info:

* proper noun - Boolean to indicate if this is a proper noun.
* occurrences - An exhaustive list of occurrences of the word. These will be RC links.

The markdown file will contain the following:

* [x] Aliases (RC links)
* [x] Examples (RC links)
* [x] Bible links can have verse granularity.
* [ ] Strong's numbers

### PR DESCRIPTION
**Pull request by neutrinog**
Fixes #41 

Re-purposed the config.yaml. Still need feedback on how to display Strong's numbers

### CODE REVIEW

File: docs/container_types.rst [hunk 1]
```diff
@@ -176,33 +176,62 @@ The dictionary terms are used as the chapter :ref:`slug` and the description of
...
+The ``01.txt`` file contains the description of the term. The term title must always be at the top of the file as a
+h1 heading (a single #). :ref:`Links <linking>` may be used to reference other words, or content in other containers.
+
+.. note:: This type is in progress. We may add Strong's numbers to the markdown as well.
 
 .. code-block:: markdown
 
-    # Aaron
+    # Aaron #
+
```
Comment (jag3773) [inline docs/container_types.rst]: I would put a "Strongs:" heading right here, that can have the strongs number.  Aaron's strongs number would be H0175 for the example.

File: docs/container_types.rst [hunk 2]
```diff
@@ -176,33 +176,62 @@ The dictionary terms are used as the chapter :ref:`slug` and the description of
...
+    Aaron was Moses' older brother. God chose Aaron to be the first high priest for the people of Israel.
+
+    * Aaron helped Moses speak to Pharaoh about letting the Israelites go free.
+    * While the Israelites were traveling through the desert, Aaron sinned by making an idol for the people to worship.
+    * God also appointed Aaron and his descendants to be the [priests](kt/priest) for the people of Israel.
+
+    (Translation suggestions: [How to Translate Names](en/ta-vol1/translate/man/translate-names))
+
+    (See also: [[kt/priest]] **·** [[other/moses]] **·** [[other/israel]])
```
Comment (jag3773) [inline docs/container_types.rst]: These should be separated by commas.  I don't know who but the dots in there but the commas are much better.

File: docs/container_types.rst [hunk 3]
```diff
@@ -176,33 +176,62 @@ The dictionary terms are used as the chapter :ref:`slug` and the description of
...
-          - aaronalias # note: not a real alias for this word
-        examples:
-          - '09-15'
-          - '10-05'
-
-Examples are tricky because a ``dict`` may be referenced by many different resources and projects.
-Therefore we cannot specify a RC link but instead must simply provide the chapter and chunk that contains the example.
-
+        proper_noun: true
+        occurrences:
```
Comment (jag3773) [inline docs/container_types.rst]: Let's add an entry here for 'false_positives; where we can list hits that a search might turn up but should not be shown by the app.

Comment (neutrinog) [inline docs/container_types.rst]: done

File: docs/container_types.rst (hunk 2 above)
Comment (neutrinog) [inline docs/container_types.rst]: done

File: docs/container_types.rst (hunk 1 above)
Comment (neutrinog) [inline docs/container_types.rst]: going with the solution you proposed in https://github.com/unfoldingWord-dev/Resource-Container/issues/41#issuecomment-294017481

**Comment by neutrinog** [comment]
@jag3773 you want to look at this again?

**Comment by PhotoNomad0** [comment]
@neutrinog - Looks good to me.  Looks like you address Jesse's concerns.  Do you want me to merge it?

**Comment by neutrinog** [comment]
It might be good for @jag3773 to look at it again in case he changed his mind again.

File: docs/container_types.rst [hunk 4]
```diff
@@ -176,33 +176,67 @@ The dictionary terms are used as the chapter :ref:`slug` and the description of
...
+    ## Examples from the Bible stories: ##
+
+    * __[09:15](en/tn/obs/help/09/15)__ God warned Moses and __Aaron__  that Pharaoh would be stubborn.
+    * __[10:05](en/tn/obs/help/10/05)__ Pharaoh called Moses and __Aaron__  and told them that if they stopped the plague, the Israelites could leave Egypt.
+    * __[13:09](en/tn/obs/help/13/09)__ God chose Moses' brother, __Aaron__, and Aaron's descendants to be his priests.
+    * __[13:11](en/tn/obs/help/13/11)__ So they (the Israelites) brought gold to __Aaron__  and asked him to form it into an idol for them!
+    * __[14:07](en/tn/obs/help/14/07)__ They (the Israelites) became angry with Moses and __Aaron__  and said, "Oh, why did you bring us to this horrible place?"
+
+
+The ``config.yaml`` file contains extra details about the term that may be helpful for some automation tools.
```
Comment (jag3773) [inline docs/container_types.rst]: Let's add two clarifying comments on what these mean.  Maybe this:

If `false_positives` exists, it is a list of places that should be excluded.  For example, if a typical regex search for "Aaron" would turn up instances that should not be shown the user, they should be listed here.

Alternatively, if the `occurrences` array exists, then it specifies the *entire* list of occurrences of this word in the given resource.  If this key exists then a regex search *should not be performed* by the software.

Comment (neutrinog) [inline docs/container_types.rst]: done

**Pull request merged**